openpype/plugins/publish/integrate_new.py

```diff
@@ -979,142 +741,104 @@ def get_rootless_path(self, anatomy, path):
+            dict: file info dictionary
         """
-        local_site = 'studio'  # default
-        remote_site = None
-        always_accesible = []
-        sync_project_presets = None
-
-        rec = {
+        return {
             "_id": ObjectId(),
-            "path": path
+            "path": self.get_rootless_path(anatomy, path),
+            "size": os.path.getsize(path),
+            "hash": openpype.api.source_hash(path),
+            "sites": sites
         }
-        if size:
-            rec["size"] = size
 
-        if file_hash:
-            rec["hash"] = file_hash
 
-        if sites:
-            rec["sites"] = sites
-        else:
-            system_sync_server_presets = (
-                instance.context.data["system_settings"]
-                                     ["modules"]
-                                     ["sync_server"])
-            log.debug("system_sett:: {}".format(system_sync_server_presets))
-
-            if system_sync_server_presets["enabled"]:
-                sync_project_presets = (
-                    instance.context.data["project_settings"]
-                                         ["global"]
-                                         ["sync_server"])
-
-            if sync_project_presets and sync_project_presets["enabled"]:
-                local_site, remote_site = self._get_sites(sync_project_presets)
-
-                always_accesible = sync_project_presets["config"]. \
-                    get("always_accessible_on", [])
-
-            already_attached_sites = {}
-            meta = {"name": local_site, "created_dt": datetime.now()}
-            rec["sites"] = [meta]
-            already_attached_sites[meta["name"]] = meta["created_dt"]
-
-            if sync_project_presets and sync_project_presets["enabled"]:
-                if remote_site and \
-                        remote_site not in already_attached_sites.keys():
-                    # add remote
-                    meta = {"name": remote_site.strip()}
-                    rec["sites"].append(meta)
-                    already_attached_sites[meta["name"]] = None
-
-                # add skeleton for site where it should be always synced to
-                for always_on_site in always_accesible:
-                    if always_on_site not in already_attached_sites.keys():
-                        meta = {"name": always_on_site.strip()}
-                        rec["sites"].append(meta)
-                        already_attached_sites[meta["name"]] = None
-
-                # add alternative sites
-                rec = self._add_alternative_sites(system_sync_server_presets,
-                                                  already_attached_sites,
-                                                  rec)
-
-            log.debug("final sites:: {}".format(rec["sites"]))
-
-        return rec
-
-    def _get_sites(self, sync_project_presets):
+class SiteSync(object):
+    """Logic for Site Sync Module functionality"""
```

Comment:
The logic defined in this temporary SiteSync class should actually be coming from the Site Sync module. @kalisp has it on its todo list to look into our options regarding exposing that as API on the Site Sync module.

But for now at least this keeps it separated from the Integrator plug-in.

Reply:
Agreed this keep it contained and we can move it in another PR